packages/typespec-ts/src/modular/emitModels.ts

```diff
@@ -166,6 +172,15 @@ export function buildModels(
     ]);
   }
 
+  const aliases = extractAliases(codeModel);
+  aliases.forEach((alias) => {
+    modelsFile.addTypeAlias({
+      name: alias.name!,
+      isExported: true,
+      docs: ["Base type for " + alias.name],
```

Comment:
I prefer to have this description when constructing this type not where we generate it.

The alias and aliasTypes are generic for me.

Reply:
We only have one model for polymorphic base, and we need to build both the Parent model type and the type alias union from that model. so description has ready been taken for Parent model, are you suggesting to add a new attribute for alias description ?

Reply:
Got the idea here, we map the one model type to two definitions (one interface + one alias). to me the confusion is we do the ad-hoc alias

Reply:
PetKind is a real type in typespec, whereas Pet is not, and our codegen framework doesn't support to emit more than one models with one single input model. Maybe we can refactor a little in the future. How do you think ?

Reply:
Sure then.